woocommerce/woocommerce-gutenberg-products-block
# Making filter block contextual on the front end

## What

- Thanks to the amazing `ProductCollectionData` route, we can make all filter blocks contextual in one place.
- In additional to the collection data params in #10873, we convert the `ProductCollection` block query context to Store API ProductCollectionData route params.

Partial fixes #10985

## Why



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Update the `$colleciton_data_params_mapping` property as follow
```
	private $collection_data_params_mapping = array(
		'calculate_price_range'         => 'woocommerce/price-filter',
		'calculate_stock_status_counts' => 'woocommerce/stock-filter',
		'calculate_attribute_counts'    => 'woocommerce/attribute-filter',
		'calculate_rating_counts'       => 'woocommerce/rating-filter',
	);
```
2. Edit line 182-184 of `CollectionFilter.php` file:
```
		if ( ! empty( $response['body'] ) ) {
			error_log( print_r( $response['body'], true ) );
			return $response['body'];
		}
```
4. On a new page, add the `Product Collection` block.
5. Apply some Filter settings, like Stock status, Product Attribute, Taxonomies.
6. Then add `Collection Filters` block to the block above.
7. Add Price filter, Stock filter, to the `Collection Filters` block.
8. Visit the page on the front end.
9. Check the log, see the data includes the correct price range that matches the Product Collection block's results.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
